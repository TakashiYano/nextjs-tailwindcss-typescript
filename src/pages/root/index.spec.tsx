import { fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";

import Root from "./index.page";

describe("Root page", () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<Root />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("clicking button triggers alert", () => {
    const { getByText } = render(<Root />);
    window.alert = jest.fn();
    fireEvent.click(getByText("Click me!"));
    expect(window.alert).toHaveBeenCalledWith("Hello, World!");
  });
});
