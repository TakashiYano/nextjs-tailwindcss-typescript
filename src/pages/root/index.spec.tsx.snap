// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root page matches snapshot 1`] = `
<div>
  <h2>
    Root
  </h2>
  <button
    className="p-2"
    onClick={[Function]}
  >
    Click me!
  </button>
</div>
`;